src/core/configureCommandFactory.js

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { queryString, stringToBoolean } from "../utils";
+import logQueryParam from "../constants/logQueryParam";
+
+export default ({
+  componentCreators,
+  createConfig,
+  coreConfigValidators,
+  logCommand,
+  logger,
+  initializeComponents,
+  setErrorsEnabled,
+  window
+}) => options => {
+  const config = createConfig(options);
+  config.addValidators(coreConfigValidators);
+  componentCreators.forEach(createComponent => {
+    const { configValidators } = createComponent;
+    config.addValidators(configValidators);
+  });
+  config.validate();
+  setErrorsEnabled(config.errorsEnabled);
+  const parsedQueryString = queryString.parse(window.location.search);
+  logCommand({
+    enabled:
+      parsedQueryString[logQueryParam] !== undefined
+        ? stringToBoolean(parsedQueryString[logQueryParam])
+        : config.logEnabled
+  });
+  // toJson is expensive so we short circuit if logging is disabled
+  if (logger.enabled) logger.log("Computed configuration:", config.toJSON());
+  return initializeComponents(config);
```

Comment:
Seems weird to pass this function in to just call it once at the end.  It's basically a callback for a sync function.

I think it'd be better to do this inside of core/main.js:
```
const configureCommand = options => initializeComponents(buildAndValidateConfig(options))
```

Reply:
Yeah, I debated doing this. Here was my mindset: I wasn't sure what the appetite was for logic inside `main.js`. Even though there's very little here, it would be the first function declared in main.js that wasn't just getting or setting (I went back and forth on those too).  If I did do this in main.js, I would want to move the `logCommandFactory.js` logic inside main.js as well so it's not a loner. I'm totally cool with this though. Still prefer it?

Reply:
I'm not sure what the guideline should be between object-creation-and-wiring-code and testable-code.  To me it feels like function composition is the same as passing in or wiring and doesn't need to be testable.  We could just as easily make a util called `compose` to do this.

Your comment about logging brings up something that has bothered me too. We have logging concerns in a bunch of different places.
* createLogController
* createLogger
* createLogCommand
* loggerToolFactory

I'm wondering if we could move all of this inside of createLogController
```
const createLogController = (instanceNamespace, createNamespacedStorage, ...) => {
  ...
  return {
    get logEnabled() {
      ...
    },
    set logEnabled(value) {
      ...
    },
    readQueryString(window) {
    },
    readConfig(config) {
    },
    logCommand(options) {
    },
    logTool() {
    },
    getCoreLogger() {
    }
}
```
After doing this exercise, I realized that we could probably put the check here to grab the query string parameter even before the configure command.  I see it is here because the query string parameter should override the config, but if we had all the logging logic together we could easily implement this functionality while still having the query string parameter take precedence.  We may even be able to incapsulate the get and set of logEnabled.

Reply:
Regarding `const configureCommand = options => initializeComponents(buildAndValidateConfig(options))`. At that point, there's not really a need for `initializeComponents` and `buildAndValidateConfig` to come from factories. For example, main.js could have this instead:

```
const configureCommand = options => {
  const config = buildAndVailidateConfig({
    config: options,
    componentCreators,
    createConfig,
    configValidators,
    logCommand,
    logger,
    initializeComponents,
    setErrorsEnabled,
    window
  });
  return initializeComponents({
    config,
    componentCreators,
    logger,
    createCookieProxy,
    createComponentNamespacedCookieJar,
    lifecycle,
    componentRegistry,
    tools: {
      config: configToolFactory(),
      cookieJar: cookieJarToolFactory(
        createCookieProxy,
        createComponentNamespacedCookieJar,
        getTopLevelDomain
      ),
      enableOptIn: enableOptInToolFactory(optIn),
      logger: loggerToolFactory(logger),
      network: networkToolFactory(createNetwork, lifecycle, logger)
    },
    optIn
  });
}
```

I'm going to create a separate ticket for this.